DemoApp/app/src/main/java/com/colortv/demo/ExoPlayerVideoActivity.java

```diff
@@ -0,0 +1,179 @@
+package com.colortv.demo;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.net.Uri;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.View;
+
+import com.colortv.android.ColorTvContentRecommendationListener;
+import com.colortv.android.ColorTvError;
+import com.colortv.android.ColorTvSdk;
+import com.google.android.exoplayer2.DefaultLoadControl;
+import com.google.android.exoplayer2.ExoPlaybackException;
+import com.google.android.exoplayer2.ExoPlayer;
+import com.google.android.exoplayer2.ExoPlayerFactory;
+import com.google.android.exoplayer2.LoadControl;
+import com.google.android.exoplayer2.SimpleExoPlayer;
+import com.google.android.exoplayer2.Timeline;
+import com.google.android.exoplayer2.extractor.DefaultExtractorsFactory;
+import com.google.android.exoplayer2.extractor.ExtractorsFactory;
+import com.google.android.exoplayer2.source.ExtractorMediaSource;
+import com.google.android.exoplayer2.source.MediaSource;
+import com.google.android.exoplayer2.source.TrackGroupArray;
+import com.google.android.exoplayer2.trackselection.AdaptiveVideoTrackSelection;
+import com.google.android.exoplayer2.trackselection.DefaultTrackSelector;
+import com.google.android.exoplayer2.trackselection.TrackSelection;
+import com.google.android.exoplayer2.trackselection.TrackSelectionArray;
+import com.google.android.exoplayer2.trackselection.TrackSelector;
+import com.google.android.exoplayer2.ui.SimpleExoPlayerView;
+import com.google.android.exoplayer2.upstream.BandwidthMeter;
+import com.google.android.exoplayer2.upstream.DataSource;
+import com.google.android.exoplayer2.upstream.DefaultBandwidthMeter;
+import com.google.android.exoplayer2.upstream.DefaultDataSourceFactory;
+import com.google.android.exoplayer2.util.Util;
+
+public class ExoPlayerVideoActivity extends Activity {
+
+    private SimpleExoPlayer simpleExoPlayer;
+    private SimpleExoPlayerView simpleExoPlayerView;
+    private String videoId;
+    private String currentPlacement = "TestContent";
+
+    private ColorTvContentRecommendationListener loadContentRecommendationListener = new ColorTvContentRecommendationListener() {
+
+        @Override
+        public void onLoaded(String placement) {
+            currentPlacement = placement;
+        }
+
+        @Override
+        public void onError(String placement, ColorTvError error) {
+            Log.e(MainActivity.class.getSimpleName(), "Error while fetching ContentRecommendation for placement " +
+                    placement + " - " + error.getErrorCode().name() + ": " + error.getErrorMessage());
+            ColorTvSdk.loadContentRecommendation(placement, videoId);
```

Comment:
Loading content recommendation on error isn't such a good idea since it will go into a cycle if the error keeps repeating.